minicluster/src/main/java/org/apache/accumulo/miniclusterImpl/MiniAccumuloClusterImpl.java

```diff
@@ -1094,6 +1098,35 @@ protected ExecutorService getShutdownExecutor() {
     return executor;
   }
 
+  public void stopProcessesWithTimeout(final ServerType type, final List<Process> procs,
+      final long timeout, final TimeUnit unit) {
+
+    final List<Future<Integer>> futures = new ArrayList<>();
+    for (Process proc : procs) {
+      futures.add(executor.submit(() -> {
+        proc.destroy();
+        proc.waitFor(timeout, unit);
+        return proc.exitValue();
+      }));
+    }
+
+    while (!futures.isEmpty()) {
+      Iterator<Future<Integer>> iter = futures.iterator();
+      while (iter.hasNext()) {
+        Future<Integer> f = iter.next();
```

Comment:
it might simplify things to do `futures.removeIf()` here although it doesn't make a functional difference.